towerofhanoi/Program.cs

```diff
@@ -0,0 +1,228 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace towerofhanoi
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Welcome to Tower of Hanoi");
+            Game towerofHanoiGame = new Game();
+            towerofHanoiGame.Play();
+        }
+    }
+    class Blocks
+    {
+        public int weightedBlock { get; set; }
+        public Blocks (int weightedBlock)
+        {
+            this.weightedBlock = weightedBlock;
+        }
+    }
+ class Towers {
+        public Stack<Blocks> towerBlocks { get; set; }
+        public Towers () {
+            this.towerBlocks = new Stack<Blocks> ();
+        }
+    }
+    class Game
+    {
+        //declaring variables and definitions
+       Dictionary<String, Towers> MakeTowers = new Dictionary<String, Towers>();
+        bool win = false;
+        string userChoice1 = "";
+        bool validMoveFrom = false;
+        String moveFrom = "";
+        String userChoice2 = "";
+        bool validMoveTo = false;
+        String movedTo = "";
+
+        //Game Constructor
+
+        public Game()
+        {
+            MakeTowers["A"] = new Towers();
+            MakeTowers["B"] = new Towers();
+            MakeTowers["C"] = new Towers();
+            //makes blocks for the towers
+            Blocks block1 = new Blocks(1);
+            Blocks block2 = new Blocks(2);
+            Blocks block3 = new Blocks(3);
+            Blocks block4 = new Blocks(4);
+            //adds blocks to tower A for the initial start of the game
+            MakeTowers["A"].towerBlocks.Push(block4);
+            MakeTowers["A"].towerBlocks.Push(block3);
+            MakeTowers["A"].towerBlocks.Push(block2);
+            MakeTowers["A"].towerBlocks.Push(block1);
+
+        }
+        public void PrintBoard()
+        {
+            foreach (String name in MakeTowers.Keys)
+            {
+                System.Console.WriteLine();
+                System.Console.WriteLine(name + ": ");
+                Towers tower = MakeTowers[name];
+                Stack<Blocks> stackofBlocks = tower.towerBlocks;
+                Stack<Blocks> stackPrinted = new Stack<Blocks>();
+
+                foreach (Blocks b in stackofBlocks)
+                {
+                    stackPrinted.Push(b);
+                    foreach (Blocks z in stackPrinted)
```

Comment:
I am not sure what you are trying to do here, but it causes each tower to be printed out multiple times. You should only need 2 foreach: one foreach to go through all towers and another to go through each block on the tower.

Reply:
I guess I wanted to make sure  that they are in order when printed